### DISCUSSION TITLE
Best way with Ethernet

### DISCUSSION THREAD

**Discussion by Wyatt59** [schreibfaul1/ESP32-audioI2S#282] [Q&A]
Hello
It's work with a lot of change.. I copy 2 files and rename it like Audioeth.h  Audioeth.cpp ,... add some line like ..#include <Ethernet.h> and change add eth in all fonctions ( AudioBuffer =>  AudioBuffereth) .
Is it the best way or there is a better thing to do????

**Comment by ghost** [comment]
Hello,
could you please share changes you have made in the original code in order to add Ethernet communication? I am looking exactly for the same. I want to build an internet radio which communicates via Ethernet (not WiFi). Thx

**Reply by Wyatt59** [reply]
ho yes , it's use a old relise (january 2022)
you can use ethernet ou wifi.

[Audioeth.zip](https://github.com/schreibfaul1/ESP32-audioI2S/files/8943164/Audioeth.zip)

and  add this :
#include "Ethernet.h"
#include "Audioeth.h"
Audioeth audioeth;


for use you need to add eht after the wifi command like this

 audioeth.i2s_mclk_pin_select(I2S_MCLK);
      audioeth.setPinout(I2S_BCLK, I2S_LRC, I2S_DSIN);
      // audioeth.forceMono(true);
      audioeth.setVolume(3); // 0...21

 audioeth.connecttohost("http://streaming.radio.rtl.fr/rtl-1-44-128");

**Reply by Wyatt59** [reply]
i use a w5500

**Reply by Wyatt59** [reply]
with esp32 audio kit

**Reply by Wyatt59** [reply]
![thumbnail_20220620_213406](https://user-images.githubusercontent.com/94049759/174667827-2a856a18-4735-48da-bed9-9db15c576cdd.jpg)

**Reply by Wyatt59** [reply]
tel me who you whant to use it (esp32 which ethernet card...sound card)

**Reply by ghost** [reply]
Thank you very much for your prompt reply. I will test it tomorrow. I will use W5500 module with PCM5102 decoder. It works with WiFi without any problem however I prefer wired connection.

**Reply by ghost** [reply]
Hello Wyatt59, I have tested your modified library for Ethernet and it works excellent! Thanks again, you saved my project. Since I am not probably the only one who would like to build internet radio with Ethernet connection based on W5500 module, I would like to post the detailed instructions here on the GitHub. Could I (with your permission) reference your modified library in my project description?

**Reply by Wyatt59** [reply]
My hour of glory !!! ho yes i give you my permission !!!
l'am in Holiday in 3week i will update my projet with the last library... (expect if you update it before :-)))

**Reply by Wyatt59** [reply]
For information before use esp32 audio kit, i use VS1053 + esp32 it's work but not some time not ... does PCM5102 Works 100%???

**Reply by ghost** [reply]
Thanks Wyatt59. Regarding VS1053 I have no experience, but PCM5102 is really amazingly reliable.

**Reply by Wyatt59** [reply]
[Audioeth.zip](https://github.com/schreibfaul1/ESP32-audioI2S/files/9173837/Audioeth.zip)
My last relise!!!

**Reply by ghost** [reply]
Many thanks, I will try it. My project is ready, I just need to post the documentation on GitHub. I have returned from a holiday today.

**Reply by ghost** [reply]
Hello Wyatt59, please see the finalized project at [](https://github.com/d7prog/Internet-Radio/blob/main/README.md). I have tested your new library, however it does not compile. Therefore I have reverted to the first version.

**Reply by Wyatt59** [reply]
hello,
it's work for me with : 
arduino ide : 1.8.19
esp32 : 2.0.4
ethernet : 2.0.1
AC101 : lastest (https://github.com/schreibfaul1/AC101)
can you send me the log ??? ( or your project ) i will test it.

**Reply by Wyatt59** [reply]
ho i see your github ... it's so clean !!! i try to compile it !!!

**Reply by ghost** [reply]
Hello Wyatt59, I have tried to compile your new Audioeth version once again and it works now! There was same problem with my Arduino IDE. After reinstall it works.

**Comment by biologist79** [comment]
I'd suggest using LAN8720 along with ESP32 as it's working almost the same way like WiFi does. Didn't test so far but I guess there's a good chance it will run just out of the box (no need to do modifications at @schreibfaul1's lib. Just have a look at this one: https://github.com/espressif/arduino-esp32/blob/master/libraries/Ethernet/examples/ETH_LAN8720/ETH_LAN8720.ino.
Downside: better don't try with jumperwires or at least don't expect it to run reliable. Best way is to solder it.
https://sautter.com/blog/ethernet-on-esp32-using-lan8720/

**Reply by ghost** [reply]
Thanks for your suggestion.
